### PR TITLE
Introduce Flake8 linter

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

**Comment by untitaker** [comment]
The linter is still failing because there are multiple references to non-existing variables.

File: xandikos/icalendar.py [hunk 1]
```diff
@@ -66,9 +66,9 @@ def calendar_prop_delta(old_component, new_component):
     for field in fields:
         old_value = old_component.get(field)
         new_value = new_component.get(field)
-        if (getattr(old_value, 'to_ical', None) is None or
-            getattr(new_value, 'to_ical', None) is None or
-            old_value.to_ical() != new_value.to_ical()):
+        if getattr(old_value, 'to_ical', None) is None or \
```
Comment (jelmer) [inline xandikos/icalendar.py]: Please keep the original formatting here; PEP8 prefers parentheses:

"The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation."

File: tox.ini [hunk 2]
```diff
@@ -6,3 +6,9 @@ envlist = py35, p36
 commands = make check
 recreate = True
 whitelist_externals = make
+
+[flake8]
+# E731: Use a def instead of lambda expr
+ignore = E731
+application-package-names = xandikos
+max-line-length=100
```
Comment (jelmer) [inline tox.ini]: Please keep this at 80, per PEP8 (I realise there's quite a few place where we go over that at the moment).

File: xandikos/web.py [hunk 3]
```diff
@@ -747,24 +750,24 @@ def main(argv):
...
-    parser.add_option("--current-user-principal",
-                      default="/user/",
-                      help="Path to current user principal. [%default]")
+    parser.add_option(
+        "-d", "--directory", dest="directory", default=None,
+        help="Directory to serve from.")
+    parser.add_option(
+        "--current-user-principal", default="/user/",
+        help="Path to current user principal. [%default]")
     parser.add_option("--autocreate",
```
Comment (jelmer) [inline xandikos/web.py]: For consistency, please move "--autocreate" to the next line as well.

File: xandikos/webdav.py [hunk 4]
```diff
@@ -143,21 +143,19 @@ class NeedsMultiStatus(Exception):
 def propstat_by_status(propstat):
     bystatus = {}
     for propstat in propstat:
-        bystatus.setdefault(
-            (propstat.statuscode, propstat.responsedescription), []).append(
-                    propstat.prop)
+        bystatus \
```
Comment (jelmer) [inline xandikos/webdav.py]: Please prefer parentheses to backlashes.

File: xandikos/webdav.py [hunk 5]
```diff
@@ -1171,8 +1165,8 @@ def _do_get(self, environ, start_response, send_body):
         (body, content_length, current_etag, content_type, content_languages) = r.render(
             accept_content_types, accept_content_languages)
         if_none_match = environ.get('HTTP_IF_NONE_MATCH', None)
-        if (if_none_match and current_etag is not None and
-            etag_matches(if_none_match, current_etag)):
+        if if_none_match and current_etag is not None and \
```
Comment (jelmer) [inline xandikos/webdav.py]: here as well

**Comment by jelmer** [comment]
I've pushed fixes for the undefined names, hopefully that should address the failures.

File: .travis.yml [hunk 6]
```diff
@@ -12,5 +12,6 @@ install:
   - sudo apt-get install -qq libneon27-dev
```
Comment (jelmer) [inline .travis.yml]: You probably want to add:

  - sudo apt-get install -qq flake8

here

**Comment by untitaker** [comment]
Fixed everything.

**Comment by untitaker** [comment]
@jelmer the leftover errors are by again unused variables where I'm not sure how to fix

**Comment by jelmer** [comment]
Thanks! I've fixed the remaining assigned but unused variables in master.

**Pull request merged**